src/main/java/io/github/railroad/Railroad.java

```diff
@@ -32,24 +33,32 @@ public void start(Stage primaryStage) {
         MinecraftVersion.load();
         ForgeVersion.load();
 
-        var welcomePane = new WelcomePane();
-        var scene = new Scene(welcomePane, 1024, 768);
-        handleStyles(scene);
+        WINDOW = primaryStage;
 
-        primaryStage.setMinWidth(scene.getWidth() + 10);
-        primaryStage.setMinHeight(scene.getHeight() + 10);
-        primaryStage.setScene(scene);
-        primaryStage.setTitle("Welcome to Railroad");
-        primaryStage.show();
+        // Calculate the primary screen size to better fit the window
+        Screen screen = Screen.getPrimary();
 
-        welcomePane.getHeaderPane().getNewProjectButton().setOnAction(event -> {
-            var newProjectPane = this.newProjectPane.updateAndGet(
-                    pane -> Objects.requireNonNullElseGet(pane, NewProjectPane::new));
+        double screenW = screen.getBounds().getWidth();
+        double screenH = screen.getBounds().getHeight();
 
-            scene.setRoot(newProjectPane);
+        double windowW = Math.max(500, Math.min(screenW * 0.75, 768));
+        double windowH = Math.max(500, Math.min(screenH * 0.75, 1024));
 
-            newProjectPane.getBackButton().setOnAction(event1 -> scene.setRoot(welcomePane));
-        });
+        // Start the welcome screen and window
+        SCENE = new Scene(new Pane(), windowW, windowH);
+
+        var welcomePane = new WelcomePane();
+        SCENE.setRoot(welcomePane);
+
+        handleStyles(SCENE);
+
+        // Open setup and show the window
+        primaryStage.setMinWidth(SCENE.getWidth() + 10);
+        primaryStage.setMinHeight(SCENE.getHeight() + 10);
+        primaryStage.setScene(SCENE);
+        primaryStage.setTitle("Railroad - 1.0.0(dev)");
+        primaryStage.show();
+        primaryStage.toFront();
```

Comment:
I believe `toFront` is redundant here as `show` on a primary stage automatically should do that.

Reply:
in reality, both isn't doing what it's supposed to be done (push the windows to front)
i was searching another method to do it but i forgot lol